examples/platforms/simulation/system.c

```diff
@@ -63,56 +64,103 @@ static void handleSignal(int aSignal)
 void otSysInit(int aArgCount, char *aArgVector[])
 {
     char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+
+    static const struct option long_options[] = {
+        {"help", no_argument, 0, OT_SIM_OPT_HELP},
+        {"sleep-to-tx", no_argument, 0, OT_SIM_OPT_SLEEP_TO_TX},
+        {"time-speed", required_argument, 0, OT_SIM_OPT_TIME_SPEED},
+        {0, 0, 0, 0},
+    };
 
     if (gPlatformPseudoResetWasRequested)
     {
         gPlatformPseudoResetWasRequested = false;
         return;
     }
 
-    if (aArgCount < 2)
+    optind = 1;
+
+    while (true)
     {
-        fprintf(stderr, "Syntax:\n    %s [--sleep-to-tx] NodeId [TimeSpeedUpFactor]\n", aArgVector[0]);
-        exit(EXIT_FAILURE);
-    }
+        int c = getopt_long(aArgCount, aArgVector, "hts:", long_options, NULL);
 
-    openlog(basename(aArgVector[argIndex++]), LOG_PID, LOG_USER);
-    setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));
+        if (c == -1)
+        {
+            break;
+        }
 
-    signal(SIGTERM, &handleSignal);
-    signal(SIGHUP, &handleSignal);
+        switch (c)
+        {
+        case OT_SIM_OPT_UNKNOWN:
+            PrintUsage(aArgVector[0], EXIT_FAILURE);
+            break;
+        case OT_SIM_OPT_HELP:
+            PrintUsage(aArgVector[0], EXIT_SUCCESS);
+            break;
+        case OT_SIM_OPT_SLEEP_TO_TX:
+            gRadioCaps |= OT_RADIO_CAPS_SLEEP_TO_TX;
+            break;
+        case OT_SIM_OPT_TIME_SPEED:
+            speedUpFactor = (uint32_t)strtol(optarg, &endptr, 10);
+            if (*endptr != '\0' || speedUpFactor == 0)
+            {
+                fprintf(stderr, "Invalid value for TimerSpeedUpFactor: %s\n", optarg);
+                exit(EXIT_FAILURE);
+            }
+            break;
+        default:
+            break;
+        }
+    }
 
-    if (!strcmp(aArgVector[argIndex], "--sleep-to-tx"))
+    if (optind != aArgCount - 1)
     {
-        gRadioCaps |= OT_RADIO_CAPS_SLEEP_TO_TX;
-        ++argIndex;
+        PrintUsage(aArgVector[0], EXIT_FAILURE);
     }
 
-    gNodeId = (uint32_t)strtol(aArgVector[argIndex], &endptr, 0);
+    gNodeId = (uint32_t)strtol(aArgVector[optind], &endptr, 0);
 
     if (*endptr != '\0' || gNodeId < 1 || gNodeId > MAX_NETWORK_SIZE)
     {
-        fprintf(stderr, "Invalid NodeId: %s\n", aArgVector[argIndex]);
+        fprintf(stderr, "Invalid NodeId: %s\n", aArgVector[optind]);
         exit(EXIT_FAILURE);
     }
 
-    ++argIndex;
-
-    if (aArgCount > argIndex)
-    {
-        speedUpFactor = (uint32_t)strtol(aArgVector[argIndex], &endptr, 0);
+    openlog(basename(aArgVector[0]), LOG_PID, LOG_USER);
+    setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));
 
-        if (*endptr != '\0' || speedUpFactor == 0)
-        {
-            fprintf(stderr, "Invalid value for TimerSpeedUpFactor: %s\n", aArgVector[argIndex]);
-            exit(EXIT_FAILURE);
-        }
-    }
+    signal(SIGTERM, &handleSignal);
+    signal(SIGHUP, &handleSignal);
```

Comment:
Are these changes relevant to the `get_opt` change?

Reply:
Yeah. The checking for `speedUpFactor` is moved into the parsing process above. The calling of `openlog` and `signal` are moved after the parsing process.